site/pages/smart-contracts/modular-account-v2/session-keys/adding-session-keys.mdx

```diff
@@ -0,0 +1,482 @@
+
+Adding scoped permissions to keys will happen via permission modules that you can pass as configuration parameters on the `installValidation` method. Permissions can be combined to fit your use case (e.g. you can limit a session key to only be able to spend 10 USDC within the next 24 hours on behalf of your account).
+
+## Adding a global session key (i.e. additional owner)
+
+This example shows:
+
+1. Adding a global session key to your Modular Account V2. This essentially gives the session key full control of your account. Functionally, this is how you can add another owner on your smart account.
+2. Adding a session key that can only call 'execute' on the account. Functionally, this allows the session key to have full control of the account other than changing the underlying the account implementation.
+
+```ts twoslash
+import { createModularAccountV2Client } from "@account-kit/smart-contracts";
+import {
+  installValidationActions,
+  getDefaultSingleSignerValidationModuleAddress,
+  SingleSignerValidationModule,
+} from "@account-kit/smart-contracts/experimental";
+import { LocalAccountSigner } from "@aa-sdk/core";
+import { sepolia, alchemy } from "@account-kit/infra";
+import { generatePrivateKey } from "viem/accounts";
+import { type SmartAccountSigner } from "@aa-sdk/core";
+
+const client = (
+  await createModularAccountV2Client({
+    chain: sepolia,
+    transport: alchemy({ apiKey: "your-api-key" }),
+    signer: LocalAccountSigner.privateKeyToAccountSigner(generatePrivateKey()),
+  })
+).extend(installValidationActions);
+
+let sessionKeyEntityId = 1;
+const ecdsaValidationModuleAddress =
+  getDefaultSingleSignerValidationModuleAddress(client.chain);
+const sessionKeySigner: SmartAccountSigner =
+  LocalAccountSigner.mnemonicToAccountSigner("SESSION_KEY_MNEMONIC");
+
+// 1. Adding a session key with full permissions
+await client.installValidation({
+  validationConfig: {
+    moduleAddress: ecdsaValidationModuleAddress,
+    entityId: sessionKeyEntityId,
+    isGlobal: true,
+    isSignatureValidation: true,
+    isUserOpValidation: true,
+  },
+  selectors: [],
+  installData: SingleSignerValidationModule.encodeOnInstallData({
+    entityId: sessionKeyEntityId,
+    signer: await sessionKeySigner.getAddress(), // Address of the session key
+  }),
+  hooks: [],
+});
+
+// 2. Adding a session key that can only call `execute` on the account
+sessionKeyEntityId = 2;
+const executeSelector = "0xb61d27f6";
```

Comment:
Could we change this to pull the ABI like so: https://github.com/alchemyplatform/aa-sdk/blob/9230c8f9c9af7a9a7d90538ef6ec42dcca1c39db/examples/ui-demo/src/hooks/useRecurringTransactions.ts#L210-L213

Rather than hard-coding the selector.

Also, it might be best practice to show adding both `execute` and `executeBatch`.

Reply:
Note: I'm not sure we correctly export this ABI, we might need to check that.